src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceDesignerAssembly.cs

```diff
@@ -0,0 +1,340 @@
+// Copyright (C) 2011 Xamarin, Inc. All rights reserved.
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.Versioning;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Android.Build.Tasks;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace Xamarin.Android.Tasks
+{
+	public class GenerateResourceDesignerAssembly : AndroidTask
+	{
+		const string DesignerAssemblyName = "Xamarin.Android.Resource.Designer";
+		public override string TaskPrefix => "GRDA";
+
+		[Required]
+		public ITaskItem RTxtFile { get; set; }
+
+		public ITaskItem ResourceMap { get; set; }
+
+		[Required]
+		public bool IsApplication { get; set; }
+
+		[Required]
+		public bool DesignTimeBuild { get; set; }
+
+		[Required]
+		public ITaskItem OutputFile { get; set; }
+
+		[Required]
+		public string TargetFrameworkVersion { get; set; }
+
+		[Required]
+		public string TargetFrameworkIdentifier { get; set; }
+
+		[Required]
+		public string ProjectDir { get; set; }
+
+		[Required]
+		public ITaskItem[] Resources { get; set; }
+
+		[Required]
+		public string ResourceDirectory { get; set; }
+
+		public ITaskItem[] AdditionalResourceDirectories { get; set; }
+
+		TypeReference intArray;
+		TypeReference intRef;
+		TypeReference objectRef;
+		Dictionary<string, string> resource_fixup = new Dictionary<string, string> (StringComparer.OrdinalIgnoreCase);
+
+		public override bool RunTask ()
```

Comment:
Why not use source generators instead of this fragile logic mimicking compiler behaviour?

Reply:
What we are trying to achieve is to generate an assembly. My understanding of source generators is that they would generate code which would then be included in the project which is being compiled (please correct me if I am wrong).
The idea here is to create a stand alone assembly which contains all the resource ids. But can be easily swapped out with the final version once we know what all the final app resource id values are.

Hope that makes sense.

Reply:
The way this works in Java is definitely weird:

1. Android Java libraries have a `R.java` -- but the class is missing from the resulting package, because the integer values are unknown at library build time. This means that Android Java libraries are actually incomplete, missing a `R.class` file.
2. In any Android application that references the Java library, an `R.java` of the appropriate package name is generated and included in the main app -- with the correct integers in place.

@marek-safar so when we try to replicate this in C#, a "reference assembly" seems like the only approach, details at: https://github.com/xamarin/xamarin-android/issues/6310

The current implementation we have swaps out values at runtime, which isn't great.

I'm unsure how we would use a Roslyn source generator here -- but it would be nice if we could somehow? It is certainly easier to generate C# than IL.

Reply:
I think this is very doable. One way I can think of is to set `EmitCompilerGeneratedFiles` property and write source generators with some if-def to not compile the files. Then in your extra task you pass `$(IntermediateOutputPath)/generated/<generator-name>` to the compiler again with the define set.

Reply:
Also, this approach would likely work for F# too